riscv-page-tables/src/page_table.rs

```diff
@@ -649,6 +659,16 @@ impl<T: FirstStagePagingMode> FirstStagePageTable<T> {
         })
     }
 
+    /// Create a new page table root from the privided `root`, with the same constraint as
+    /// `FirstStagePageTable::new()`. Copy the content of the root page table from `src`.
+    ///
+    /// Note: The PTEs will be shared between `src` and the new page table created.
```

Comment:
This seems incredibly unsafe to me. You're basically requiring that any existing non-root PTs and PTEs remain static after this point, otherwise there's unguarded shared mutability. I get that you plan on using this in a safe way -- by putting the per-CPU mappings far away from everything else in VA space -- but this still seems like a massive foot-gun.

Unless you want to go down the route of somehow enforcing that this shared range of VA space doesn't get modified (e.g. PTE locks, or "sealing" mappings), I'd just make a deep copy like in your previous iteration for now.

Reply:
Yes, I'd like to go towards enforcing this. In general there's an advantage to be had in having page table extremely static, set once. I think that mapping page tables read-only once set is a decent way to go.
I don't think salus has any advantage in having dynamic paging.

Reply:
Just to get back at this discussion, I think a better solution here is to create some form of type safety (something similar to `VmPages`, with an `InitializingVmPages` state that allows wider changes, and a state of `FinalizedVm` that prevents that). Once this is done, it becomes irrelevant whether we have a a full copy of page tables or a shared one.

Reply:
Created HypMap in new version. Also switched to full copies to avoid confusing rust's model.